website/docs/learning-and-resources/how-to-guides/how-to-work-with-local-apis-on-appsmith/README.mdx

```diff
@@ -0,0 +1,223 @@
+
+:::caution note
+**Linux**: If your containers are running on a Linux machine, start your Appsmith container with the following command line argument to `docker run`. This is only supported for Docker 20.10.0 and later.
+
+```bash
+--add-host=host.docker.internal:host-gateway
+```
+
+Or, for `docker-compose`, update your docker-compose.yaml file to include the following `extra_hosts` option:
+
+```yml
+services: ...
+  postgres: ...
+    environment: ...
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
+```
+:::
+
+<figure>
+  <img src="/img/local_db_config.png" style={{width: "100%", height: "auto"}} alt="Connecting from Appsmith in Docker container to PostgreSQL database running on localhost" />
+  <figcaption align="center"><i>Connecting from Appsmith in Docker container to PostgreSQL database running on localhost</i></figcaption>
+</figure>
+
+
+</TabItem>
+
+<TabItem label="Docker network" value="docker-network">
+
+
+### Datasource in Docker
+
+This guide describes how to connect your APIs and databases to your Appsmith instance when they are each in separate Docker containers.
+
+#### Set up services
+
+In this guide, you'll use networked Docker containers to spin up a Postgres instance alongside Appsmith and connect to the database from an Appsmith app.
+
+1. From the command line, create a Docker network with the following:
+
+    ```bash
+    docker network create appsmithnetwork
+    ```
+
+1. Create two folders, `Appsmith` and `Datasource`.
+
+1. Within the `Appsmith` folder, create a `docker-compose.yml` file for Appsmith. Assign it to the network from the first step, such as in the bottom of this example file:
+
+    ```yaml
+    version: "3"
+    services:
+        appsmith:
+            # appsmith-ce for community edition, appsmith-ee for business.
+            image: index.docker.io/appsmith/appsmith-ce:v1.9.12
+            container_name: appsmith
+            ports:
+                - "80:80"
+                - "443:443"
+            volumes:
+                - ./stacks:/appsmith-stacks
+            restart: unless-stopped
+            networks:
+                - appsmithnetwork
+    networks:
+    appsmithnetwork:
+        external: true
+    ```
+
+1. Start the Appsmith container with:
+
+    ```bash
+    docker-compose up -d
+    ```
+
+1. Inside the `DataSource` folder, create `docker-compose.yml` file and assign it to the same network. For example:
+
+    ```yaml
+    version: '3'
+    services:
+        postgres:
+            image: 'postgres:12'
+            restart: always
+            volumes:
+                - './postgres_data:/var/lib/postgresql/data'
+            environment:
+                - POSTGRES_PASSWORD=<your-postgres-password>
+            ports:
+                - '5433:5432'
+            networks:
+                - appsmithnetwork
+    networks:
+        appsmithnetwork:
+            external: true
+    ```
+
```

Comment:
I believe it would be relevant to include a note emphasizing that the important aspect of these two docker-compose files is the networking and the configuration of those networks.